Mathlib/Topology/Algebra/Module/WeakDual.lean

```diff
@@ -326,6 +353,74 @@ theorem coe_map (f : E →L[𝕜] F) : (WeakSpace.map f : E → F) = f :=
   rfl
 #align weak_space.coe_map WeakSpace.coe_map
 
+/-- The canonical preimage in `E` of an open set in `WeakSpace 𝕜 E` is open in `E`. -/
+theorem isOpen_of_isOpen_WeakSpace (U : Set (WeakSpace 𝕜 E))
+    (hU : IsOpen[(WeakSpace.instTopologicalSpace : TopologicalSpace (WeakSpace 𝕜 E))] U) :
+    IsOpen ((continuousLinearMapToWeakSpace 𝕜 E) ⁻¹' U) := by
+  exact ((continuousLinearMapToWeakSpace 𝕜 E).cont).isOpen_preimage U hU
+
+/-- The canonical map from `WeakSpace 𝕜 E` to `E` is an open map. -/
+theorem isOpenMap_inv_toWeakSpace : IsOpenMap (toWeakSpace 𝕜 E).symm := by
+  apply IsOpenMap.of_inverse (continuousLinearMapToWeakSpace 𝕜 E).cont _ _
+  intro x
+  simp only [AddHom.toFun_eq_coe, LinearMap.coe_toAddHom, ContinuousLinearMap.coe_coe,
+    continuousLinearMapToWeakSpace_eq_toWeakSpace, LinearEquiv.symm_apply_apply]
+  intro x
+  simp only [AddHom.toFun_eq_coe, LinearMap.coe_toAddHom, ContinuousLinearMap.coe_coe,
+    continuousLinearMapToWeakSpace_eq_toWeakSpace, LinearEquiv.apply_symm_apply]
+
+/-- A set in `E` which is open in the weak topology is open. -/
+theorem isOpen_of_isOpen_WeakSpace' (V : Set E)
+    (hV : IsOpen[(WeakSpace.instTopologicalSpace)]
+    ((continuousLinearMapToWeakSpace 𝕜 E) '' V : Set (WeakSpace 𝕜 E))) : IsOpen V := by
+  have : (continuousLinearMapToWeakSpace 𝕜 E) ⁻¹' ((continuousLinearMapToWeakSpace 𝕜 E) '' V)
+  = V := by
+   ext x
+   constructor
+   · intro hx
+     obtain ⟨y, hy⟩ := hx
+     rw [continuousLinearMapToWeakSpace_eq_toWeakSpace,
+       continuousLinearMapToWeakSpace_eq_toWeakSpace] at hy
+     have : y = x := LinearEquiv.injective (toWeakSpace 𝕜 E) hy.2
+     rw [this] at hy
+     exact hy.1
+   · intro hx
+     simp only [Set.mem_preimage, Set.mem_image]
+     use x
+  rw [← this]
+  exact isOpen_of_isOpen_WeakSpace ((continuousLinearMapToWeakSpace 𝕜 E) '' V) hV
+
+/-- If `f : α → E` is convergent in the original topology,
+then it is convergent in the weak topology. -/
+theorem tendsto_WeakSpace_of_tendsto
+    {α : Type*} {l : Filter α} {f : α → E} {p : E} (hf : Tendsto f l (nhds p)) :
+    Tendsto (fun x ↦ (continuousLinearMapToWeakSpace 𝕜 E) (f x))
+    l (nhds ((continuousLinearMapToWeakSpace 𝕜 E) p)) := by
+  refine tendsto_nhds.mpr ?_
+  intro U hpU hU
+  rw [tendsto_nhds] at hf
+  have : (fun x ↦ (continuousLinearMapToWeakSpace 𝕜 E) (f x))
+    = (fun x ↦ ((continuousLinearMapToWeakSpace 𝕜 E) ∘ f) x) := by
+    rfl
+  rw [this, Set.preimage_comp]
```

Comment:
Moritz, I only half agree with this. While it's true that it's *acceptable*, I would say it's still discouraged, because it generally means that didn't find the right lemma, or the right lemma doesn't exist and it should be written. In this case, the right lemma is `Function.comp_apply`.

Reply:
this is no longer needed as it was a part of a theorem now removed